### DISCUSSION TITLE
Should we make unquoted string as glob pattern?

### DISCUSSION THREAD

**Discussion by hardfau1t** [nushell/nushell#9810] [Ideas]
For any command(internal or external) treat unquoted parameter as glob pattern and expand it before eval. Then remove glob from signature ( or better would be wherever  filepath is present, treat unquoted string as glob).  bash and other shells also does the same thing, even for external commands nushell also does the same thing i assume.

By doing this we can eliminate issues like #9790 . Considering `rm` command it has signature of `filepath` as positional arguments and during command eval, `rm` internally expands this, because of this `rm a[b]` and `rm "a[b]"` both are treated as glob so it fails to remove file `a[b]`

Above may be inaccurate(Its first time i went through sourcecode of nushell) or may be there are better solution for this problem. That being said I am happy to make these changes with some help

**Comment by fdncred** [comment]
We're interested in a fix for these type of issues. We have to be aware that `[ ]` is a list in nushell, unrelated to globs. I've hacked around the problem with `[]` in `ls` by separating the path part from the glob part, glob_with vs glob_from iirc, but something needs to be done for other commands.

Also complicating this issue is that we'd like to build in support for coreutils/uutils which I've started a PR for already. They have their own way of working with this type of thing.
